SignalServiceKit/src/Account/TSPreKeyManager.m

```diff
@@ -90,7 +90,8 @@ + (dispatch_queue_t)prekeyQueue
 
 + (void)checkPreKeysIfNecessary
 {
-    OWSAssert([UIApplication sharedApplication].applicationState == UIApplicationStateActive);
+    // FIXME SHARINGEXTENSION
+    //    OWSAssert([UIApplication sharedApplication].applicationState == UIApplicationStateActive);
```

Comment:
Proposal: We add a trivial global to SSK "application type", enum values: main app, app extension.  And we add a condition in a case like this.  And if we need to go further we add an `ApplicationContext` interface and the main app's AppDelegate or the sharing extension's main class implement that interface and register themselves.

This is where I foresee some kind